src/layouts/AssetManager/components/AssetControlCenter/components/sections/SecurityIdentifiersSection.tsx

```diff
@@ -57,21 +60,38 @@ export const SecurityIdentifiersSection: React.FC<
     }
   };
 
-  const handleRemoveSecurityIdentifier = async (identifierId: string) => {
-    if (!asset?.details?.assetIdentifiers) {
-      return;
-    }
-
+  const handleRemoveSecurityIdentifier = (identifierId: string) => {
     // Parse identifierId to get type and value
     const [type, value] = identifierId.split('|');
 
+    const identifierFound = securityIdentifiers.find(
+      (identifier) => identifier.type === type && identifier.value === value,
+    );
+
+    if (identifierFound) {
+      setIdentifierToDelete(identifierFound);
+      setDeleteConfirmModalOpen(true);
+    }
+  };
+
+  const confirmRemoveSecurityIdentifier = async () => {
+    if (!asset?.details?.assetIdentifiers || !identifierToDelete) {
+      return;
+    }
+
     // Filter out the identifier to remove
     const remainingIdentifiers = asset.details.assetIdentifiers.filter(
-      (identifier) => !(identifier.type === type && identifier.value === value),
+      (identifier) =>
+        !(
+          identifier.type === identifierToDelete.type &&
+          identifier.value === identifierToDelete.value
+        ),
     );
 
     // Call the asset action to update identifiers
     await modifyAssetIdentifiers(remainingIdentifiers);
+    setDeleteConfirmModalOpen(false);
+    setIdentifierToDelete(null);
   };
```

Comment:
Suggest to set the identifierToDelete to null when the transaction is running so the modal closes
```suggestion
    await modifyAssetIdentifiers(remainingIdentifiers, () => {
      setIdentifierToDelete(null);
    });
    setDeleteConfirmModalOpen(false);
  };
```